downstairs/src/lib.rs

```diff
@@ -813,7 +813,7 @@ where
                         flush_number,
                         // TODO(matt) this is a repair ID, not a job ID, but is
                         // only used for logging, so ¯\_(ツ)_/¯
```

Comment:
Should we update this comment, and, or, why does only this one  location gets the update 
for`JobId(repair_id)` ?  There are `repair_id`'s for `Message::ExtentClose`, `Message::ExtentRepair`,
and `Message::ExtentReopen` but none of them need this translation?

Reply:
`Region::reopen_extent`, `Region::repair_extent`, and `Region::close_extent` all don't take a `JobId` argument; `Region::region_flush_extent` is unique in that respect!

(`Region::region_flush_extent` then calls into `Extent::flush`, passing that `JobId`, which is used as an ID in DTrace probes)

Reply:
Alright, What do you think about 73b141ab7b7a04fdbb496ee89b13d0ed2df58fd7 ?  It's still strongly typed, but no longer relies on punning the inner `u64`.